sql-plugin/src/main/scala/org/apache/spark/sql/rapids/AggregateFunctions.scala

```diff
@@ -375,6 +375,55 @@ case class GpuSum(child: Expression, resultType: DataType)
     Aggregation.sum().onColumn(inputs.head._2)
 }
 
+case class GpuPivotFirst(
+  pivotColumn: Expression,
+  valueColumn: Expression,
+  pivotColumnValues: Seq[Any],
+  mutableAggBufferOffset: Int = 0,
+  inputAggBufferOffset: Int = 0) extends GpuDeclarativeAggregate {
```

Comment:
so this extends `GpuDeclarativeAggregate`, but the meta expression is for imperative ones. Which seems odd. If both imperative and declarative Spark CPU aggregates can be implemented by our same interface (so far anyway), perhaps we should rename ours `GpuAggregate` or something like that and use it for both kinds.